src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -166,7 +178,33 @@ private SyntaxNode Visit(SyntaxNode namedTypeNode, INamedTypeSymbol namedType)
             return namedTypeNode;
         }
 
-        private IEnumerable<INamespaceSymbol> EnumerateNamespaces(IAssemblySymbol assemblySymbol)
+        private SyntaxNode GenerateAssemblyAttributes(IAssemblySymbol assembly, SyntaxNode compilationUnit)
+        {
+            foreach (var attribute in assembly.GetAttributes())
+            {
+                compilationUnit = _syntaxGenerator.AddAttributes(compilationUnit, _syntaxGenerator.Attribute(attribute)
+                    .WithTrailingTrivia(SyntaxFactory.LineFeed));
+            }
+            return compilationUnit;
+        }
+
+        private SyntaxNode GenerateForwardedTypeAssemblyAttributes(IAssemblySymbol assembly, SyntaxNode compilationUnit)
+        {
+            foreach (INamedTypeSymbol symbol in assembly.GetForwardedTypes())
+            {
+                if (symbol.TypeKind != TypeKind.Error)
```

Comment:
If assembly references are passed in for the current IAssemblySymbol, please log a warning if TypeKind == Error.

Reply:
Added warning.
> _logger.LogWarning(string.Format("Could not find matching assembly: '{0}' in any of the search directories.", $"{symbol.ContainingAssembly.Name}.dll"));

I will introduce Resource file in follow up PR.

Reply:
Ok. The error message should not use "search directories" as in the new GenAPI we in most cases don't use directories for assembly references anymore and instead pass them in directly.

Maybe something like: `Could not resolve type '{0}' in containing assembly '{1}' via type forward. Make sure that the assembly is provided as a reference and contains the type.`

Reply:
Thanks, updated